### ISSUE TITLE
X-Art scraper issues

### ISSUE THREAD

**Issue by FlashSpazzbo** [stashapp/CommunityScrapers#1843]
**Scraper name**
X-Art

**Scraper method**
Scene, scrape with URL

**Sample URL**
https://www.x-art.com/videos/crazy_for_stacy/

## Issue 1
Scraper fails without `www` sub-domain in URL.

### Description
<details>
Scraper should change URLs that don't contain `www` sub-domain to include sub-domain.

<br>For example, `https://x-art.com/videos/crazy_for_stacy/`, should be changed to `https://www.x-art.com/videos/crazy_for_stacy/
`

Website is an inconsistent mess and browsers will sometimes not provide the `www` URL on copy-paste actions.
</details>

## Issue 2
Scraper can't scrape members' site URLs

### Description
<details>
<p>Scraper should probably modify members' site URLs to public-facing URLs.</p>

The members' site URL for the sample URL above is `https://www.x-art.com/members/videos/crazy_for_stacy/
`. Dropping the `members` sub-folder will direct to the public-facing page.

<p>The public-facing pages also have relatively cleaner HTML & CSS. The members' scene pages also don't have the scenes' dates.</p>
</details>

## Issue 3
Date field not being scraped properly.

### Description
<details>
Scraper appears to be selecting first child `h2` for the date field, instead of the unclassed `h2` element the date information resides in.

<br>Website appears to use `h2` elements for visual formatting, instead of for semantic purposes.  In sample URL, there are seven unclassed ID-less `h2` elements in the main content section, the first four of which contain content for the scene description.

The date field, based on the last five or so scenes, appears to be in front of a `h2 + span`, such as this from the sample URL:
```html
<h2>May 09, 2024 </h2>
<h2>
    <span>featuring</span>
    <a href="https://www.x-art.com/models/stacy">Stacy</a>
</h2>
``` 

<p><strong>Note that the anchor tag is not always included.</strong></p>
</details>

## Issue 4
Header elements not being scraped as scene description text.

### Description

<details><summary>Details</summary>
In the sample URL above, anything not a paragraph element is being ignored. The full description text is:

```html
<h1><strong>WE ARE BACK.</strong></h1>
<h2>This 4K VIDEO is our second little teaser - letting you know that we are once again posting new content on our site... and quite a teaser it is!&nbsp;</h2>
<p>Model Stacy Cruz shows off every little bit of her slender form in this delightful video. She loves to show off for the camera - and for YOU!</p>
<h2>Stacy is getting warmed up for her next update and video, in which a very lucky gentleman dives deeper still into the pleasures of her flesh.</h2>
<p>Sit back and enjoy and Stacy disrobes, then lets her fingers 'do the walking' across her smooth skin, up and down her absolutely heavenly body.</p>
<p>You're going to love the ending of this vidieo - plenty of closeups as Stacy brings herself to the edge of orgasm long enough to make her eyes (and ours) roll back in ecstasy. Enjoy yourself!</p>
<h2><strong>XOXO</strong></h2>
<h2>Colette (... and Brig)</h2>
``` 

The scraper is instead returning the following:
`Model Stacy Cruz shows off every little bit of her slender form in this delightful video. She loves to show off for the camera - and for YOU! Sit back and enjoy and Stacy disrobes, then lets her fingers 'do the walking' across her smooth skin, up and down her absolutely heavenly body. You're going to love the ending of this vidieo - plenty of closeups as Stacy brings herself to the edge of orgasm long enough to make her eyes (and ours) roll back in ecstasy. Enjoy yourself!`
</details> 

### Description

<details><summary>Details</summary>

<p>Scraper not respecting paragraph breaks.</p>

<p>See description for Issue 4 for example</p>
</details>

**Comment by MortonBridges** [comment]
@FlashSpazzbo I can't reproduce item 1.  The URL https://x-art.com/videos/xart_sex_to_the_rescue/ scrapes fine in my local Stash.  Can you provide a URL that reproduces the problem?

**Comment by MortonBridges** [comment]
Items 2-5 are addressed in PR1859.

**Comment by Maista6969** [comment]
A very well written bug report and a thorough fix by MortonBridges, thank you both! Closing as solved in #1859
